app/javascript/app/components/table/table-component.jsx

```diff
@@ -58,9 +58,11 @@ class SimpleTable extends PureComponent {
     if (!data.length) return null;
     const hasColumnSelectedOptions = hasColumnSelect && columnsOptions;
     const activeColumnNames = activeColumns.map(c => c.value);
-    const columnData = activeColumnNames
-      .filter(c => firstColumnHeaders.includes(c))
-      .concat(difference(activeColumnNames, firstColumnHeaders));
+    let columnData = [];
```

Comment:
what was wrong with `filter` function?

Reply:
It was just adding the firstColumns first. Not in the same order that they were passed. This is a quite sensible prop as I want to make it relatively generic

Reply:
huh, I thought `filter` was preserving the order. ok then 👍